tee-worker/service/src/prometheus_metrics.rs

```diff
@@ -244,6 +257,74 @@ fn handle_stf_call_request(req: RequestType, time: f64) {
 	observe_execution_time(category, label, time)
 }
 
+// This function will handle the conversion of the caller into a formatted string.
+fn format_caller(caller_option: Option<Identity>) -> String {
+	if let Some(caller) = caller_option {
+		// Note: This unwrap should be unfallable technically
+		return caller.to_account_id().unwrap().to_string()
+	}
+	"0xDeadBeef".to_string()
+}
+
+// This function will increment the metric with provided label values.
+fn inc_succesfull_trusted_operation_metric(operation: &str, block_str: &str, caller: &str) {
+	ENCLAVE_SUCCESFUL_TRUSTED_OPERATION
+		.with_label_values(&[operation, block_str, caller])
+		.inc();
+}
+
+fn inc_failed_trusted_operation_metric(operation: &str, block_str: &str, caller: &str) {
+	ENCLAVE_FAILED_TRUSTED_OPERATION
+		.with_label_values(&[operation, block_str, caller])
+		.inc();
+}
+
+fn handle_trusted_operation<F>(call: TrustedCall, record_metric_fn: F)
+where
+	F: Fn(&str, &str, &str),
+{
+	let block_str = format!("{}", ENCLAVE_SIDECHAIN_BLOCK_HEIGHT.get());
+	match call {
+		TrustedCall::link_identity(caller, ..) => {
+			record_metric_fn("link_identity", &block_str, &format_caller(Some(caller)));
```

Comment:
How many gauges we will end up having ?
Looks like using mixture of caller and block will make it grow pretty fast and it's hard for me to imagine the upper limit, am i right ? 🙃

Reply:
Yeah that is true, It would grow very quickly. I was looking into how we can prune it every x blocks or something, so that it is only stored on the prometheus database and not in the Enclave.

Reply:
So yeah after thinking about this, I've decided not to go with `block` and `caller`, as each label creates a new series, and since there can x and y amounts of block and caller, the gauge would exponentially increase especially at scale. So for now the counter will be used to only identify the `call` type, allowing us to know which `TrustedOperation` is requested most and few other metrics that might be important for the business as well as for us parachain devs monitoring the system for alerts etc. We can include `block` and `caller` once we find a way to emit events using prometheus library, that is it emits and forgets and prometheus DB is responsible for storing it.